distributed/semaphore.py

```diff
@@ -372,6 +372,10 @@ def __init__(self, max_leases=1, name=None, client=None, register=True):
             / 5
         )
         self._refreshing_leases = False
+
+        # Ensure that the PC below takes the correct IOLoop
+        if not self.client.io_loop.is_current:
+            self.client.io_loop.make_current()
```

Comment:
Indeed, the `add_callback` seems to work. I'm wondering how other PCs are affected and if we should replace all `pc.start()` with this.

> Should the refresh_callback be started on acquire, which is already a coroutine?

Theoretically, that should work but for whatever reason a few of the tests break when trying this. Instead of debugging this, I'd prefer to use the `add_callback`

Reply:
> Indeed, the add_callback seems to work. I'm wondering how other PCs are affected and if we should replace all pc.start() with this.

I think that we're usually pretty careful with these, but I wouldn't be surprised if something else has slipped.